PreCheck/doxygen-man/man3/SH_DatabaseContentQuestionState.3
.TH "SH_DatabaseContentQuestionState" 3 "Mardi Juillet 2 2013" "Version 0.4" "PreCheck" \" -*- nroff -*-
.ad l
.nh
.SH NAME
SH_DatabaseContentQuestionState \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <SH_DatabaseContentQuestionState\&.h>\fP
.PP
Est dérivée de \fBSH_QuestionState\fP\&.
.SS "Connecteurs publics"

.in +1c
.ti -1c
.RI "virtual void \fBsetVisibility\fP (bool isVisible)"
.br
.in -1c
.SS "Signaux"

.in +1c
.ti -1c
.RI "void \fBanswerInvalid\fP ()"
.br
.RI "\fIanswerInvalid \fP"
.ti -1c
.RI "void \fBanswerValid\fP ()"
.br
.RI "\fIanswerValid \fP"
.ti -1c
.RI "void \fBdisplayChoiceList\fP (QVariantList choices)"
.br
.ti -1c
.RI "void \fBgoNext\fP ()"
.br
.ti -1c
.RI "void \fBnext\fP ()"
.br
.ti -1c
.RI "void \fBresendInput\fP (QVariant \fBinput\fP)"
.br
.ti -1c
.RI "void \fBsendOutput\fP (QVariant \fBoutput\fP)"
.br
.in -1c
.SS "Fonctions membres publiques"

.in +1c
.ti -1c
.RI "\fBSH_DatabaseContentQuestionState\fP (QString question, QString \fBname\fP, QString databaseTable, QString tableField, QString databaseCondition='', \fBQState\fP *parent=0)"
.br
.ti -1c
.RI "bool \fBcheckValidity\fP ()"
.br
.ti -1c
.RI "QVariantList \fBchoiceList\fP ()"
.br
.ti -1c
.RI "void \fBdisplay\fP (bool canDisplay)"
.br
.ti -1c
.RI "void \fBemitResendInput\fP ()"
.br
.ti -1c
.RI "void \fBemitSendOutput\fP ()"
.br
.ti -1c
.RI "virtual QVariant \fBgivenAnswer\fP () const "
.br
.ti -1c
.RI "virtual QVariant \fBinput\fP () const "
.br
.ti -1c
.RI "virtual bool \fBisAnswerValid\fP (const QVariant &\fBgivenAnswer\fP)"
.br
.ti -1c
.RI "bool \fBisRunning\fP ()"
.br
.ti -1c
.RI "virtual QString \fBoutput\fP () const "
.br
.ti -1c
.RI "virtual QVariant \fBrawInput\fP () const "
.br
.ti -1c
.RI "virtual void \fBsetGivenAnswer\fP (const QVariant &\fBgivenAnswer\fP)"
.br
.ti -1c
.RI "virtual void \fBsetInput\fP (const QVariant &\fBinput\fP)"
.br
.ti -1c
.RI "void \fBsetOutput\fP (const QString &\fBoutput\fP)"
.br
.ti -1c
.RI "QString \fBtoString\fP ()"
.br
.ti -1c
.RI "bool \fBvisibility\fP ()"
.br
.in -1c
.SS "Fonctions membres protégées"

.in +1c
.ti -1c
.RI "void \fBonEntry\fP (QEvent *event)"
.br
.ti -1c
.RI "void \fBonExit\fP (QEvent *event)"
.br
.in -1c
.SS "Attributs privés"

.in +1c
.ti -1c
.RI "QVariantList \fBm_choices\fP"
.br
.RI "\fIm_choices \fP"
.ti -1c
.RI "bool \fBm_choicesDisplayed\fP"
.br
.RI "\fIm_choicesDisplayed \fP"
.ti -1c
.RI "QString \fBm_condition\fP"
.br
.RI "\fIm_condition \fP"
.ti -1c
.RI "QString \fBm_field\fP"
.br
.RI "\fIm_field \fP"
.ti -1c
.RI "QString \fBm_table\fP"
.br
.RI "\fIm_table \fP"
.in -1c
.SH "Description détaillée"
.PP 
Définition à la ligne 10 du fichier SH_DatabaseContentQuestionState\&.h\&.
.SH "Documentation des constructeurs et destructeur"
.PP 
.SS "SH_DatabaseContentQuestionState::SH_DatabaseContentQuestionState (QStringquestion, QStringname, QStringdatabaseTable, QStringtableField, QStringdatabaseCondition = \fC''\fP, \fBQState\fP *parent = \fC0\fP)"

.PP
Définition à la ligne 10 du fichier SH_DatabaseContentQuestionState\&.cpp\&.
.PP
Références SH_SqlDataModel::datas(), SH_MessageManager::debugMessage(), SH_SqlDataModel::fetch(), m_condition, m_field, et m_table\&.
.PP
.nf
10                                                                                                                                                                                      :
11     SH_QuestionState(question, name, parent), m_table(databaseTable), m_condition(databaseCondition), m_field(tableField)
12 {
13     SH_MessageManager::debugMessage(QString("multiple choice list with datas from %1!")\&.arg(databaseTable));
14     SH_SqlDataModel *sqlDatas = new SH_SqlDataModel();
15     QStringList fields;
16     fields << "ID" << m_field;
17     fields\&.removeDuplicates();
18     sqlDatas->fetch(m_table, m_condition, "", fields);
19     QVariantMap results = sqlDatas->datas();
20     QVariantList idValues = results\&.values("ID");
21     QVariantList fieldsValues = results\&.values(m_field);
22     for(int i = 0; i < idValues\&.length(); i++) {
23         SH_MessageManager::debugMessage(QString("new choice %1: %2")\&.arg(idValues\&.at(i)\&.toString())\&.arg(fieldsValues\&.at(i)\&.toString()));
24         //m_choices\&.insert(idValues\&.at(i)\&.toInt(), fieldsValues\&.at(i));
25     }
26 }
.fi
.SH "Documentation des fonctions membres"
.PP 
.SS "void SH_QuestionState::answerInvalid ()\fC [signal]\fP, \fC [inherited]\fP"

.PP
answerInvalid 
.PP
Référencé par SH_QuestionState::checkValidity(), SH_BillingCreationStateMachine::SH_BillingCreationStateMachine(), et SH_ServiceCharging::SH_ServiceCharging()\&.
.SS "void SH_QuestionState::answerValid ()\fC [signal]\fP, \fC [inherited]\fP"

.PP
answerValid 
.PP
Référencé par SH_QuestionState::checkValidity(), et SH_ServiceCharging::SH_ServiceCharging()\&.
.SS "SH_QuestionState::checkValidity ()\fC [inherited]\fP"

.PP
\fBRenvoie:\fP
.RS 4
bool 
.RE
.PP

.PP
Définition à la ligne 17 du fichier SH_QuestionState\&.cpp\&.
.PP
Références SH_QuestionState::answerInvalid(), SH_QuestionState::answerValid(), SH_QuestionState::givenAnswer(), SH_GenericState::goNext(), SH_MessageManager::infoMessage(), SH_QuestionState::isAnswerValid(), SH_GenericState::isRunning(), SH_InOutState::output(), SH_InOutState::setInput(), et SH_GenericState::toString()\&.
.PP
Référencé par SH_QuestionState::setGivenAnswer()\&.
.PP
.nf
18 {
19     bool ok = false;
20     if(isRunning()) {
21         ok = this->isAnswerValid(this->givenAnswer());
22         if(ok) {
23             SH_MessageManager::infoMessage(QString("%1 is a VALID answer for question '%2'!")\&.arg(this->givenAnswer()\&.toString())\&.arg(this->output()));
24             SH_InOutState::setInput(this->givenAnswer());
25             emit answerValid();
26             emit goNext();
27         } else {
28             SH_MessageManager::infoMessage(QString("%1 is an INvalid :-( answer for question '%2'")\&.arg(this->givenAnswer()\&.toString())\&.arg(this->output()));
29             emit answerInvalid();
30         }
31     }
32     return ok;
33 }
.fi
.SS "QVariantList SH_DatabaseContentQuestionState::choiceList ()"

.PP
\fBRenvoie:\fP
.RS 4
QMap<int, QVariant> 
.RE
.PP

.PP
Définition à la ligne 59 du fichier SH_DatabaseContentQuestionState\&.cpp\&.
.PP
Références m_choices, et m_choicesDisplayed\&.
.PP
.nf
59                                                          {
60     if(m_choicesDisplayed) {
61         return m_choices;
62     }
63     return QVariantList();
64 }
.fi
.SS "SH_InOutState::display (boolcanDisplay)\fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIcanDisplay\fP 
.RE
.PP

.PP
Définition à la ligne 80 du fichier SH_IOState\&.cpp\&.
.PP
Références SH_InOutState::emitSendOutput(), et SH_InOutState::m_display\&.
.PP
.nf
81 {
82         m_display=canDisplay;
83         emitSendOutput();
84 }
.fi
.SS "SH_DatabaseContentQuestionState::displayChoiceList (QVariantListchoices)\fC [signal]\fP"

.PP
Référencé par SH_InOutStateMachine::addIOState(), et setOutput()\&.
.SS "void SH_InOutState::emitResendInput ()\fC [inherited]\fP"

.PP
Définition à la ligne 92 du fichier SH_IOState\&.cpp\&.
.PP
Références SH_GenericState::isRunning(), SH_InOutState::m_input, SH_InOutState::m_isVisible, et SH_InOutState::resendInput()\&.
.PP
Référencé par SH_InOutState::setInput(), et SH_InOutState::SH_InOutState()\&.
.PP
.nf
92                                     {
93     if(isRunning() && m_isVisible) {
94         emit resendInput(m_input);
95     }
96 }
.fi
.SS "void SH_InOutState::emitSendOutput ()\fC [inherited]\fP"

.PP
Définition à la ligne 86 du fichier SH_IOState\&.cpp\&.
.PP
Références SH_GenericState::isRunning(), SH_InOutState::m_display, SH_InOutState::m_isVisible, SH_InOutState::m_output, et SH_InOutState::sendOutput()\&.
.PP
Référencé par SH_InOutState::display(), SH_InOutState::setOutput(), et SH_InOutState::SH_InOutState()\&.
.PP
.nf
86                                    {
87     if(isRunning() && m_display && !m_output\&.isEmpty() && m_isVisible) {
88         emit sendOutput(QVariant(m_output));
89     }
90 }
.fi
.SS "SH_QuestionState::givenAnswer () const\fC [virtual]\fP, \fC [inherited]\fP"

.PP
\fBRenvoie:\fP
.RS 4
QVariant 
.RE
.PP

.PP
Définition à la ligne 48 du fichier SH_QuestionState\&.cpp\&.
.PP
Références SH_QuestionState::m_givenAnswer\&.
.PP
Référencé par SH_QuestionState::checkValidity(), rawInput(), SH_QuestionState::setGivenAnswer(), SH_QuestionState::setInput(), et SH_BillingCreationStateMachine::SH_BillingCreationStateMachine()\&.
.PP
.nf
49 {
50     return this->m_givenAnswer;
51 }
.fi
.SS "SH_GenericState::goNext ()\fC [signal]\fP, \fC [inherited]\fP"

.PP
Référencé par SH_QuestionState::checkValidity(), SH_ConfirmationState::confirmInput(), SH_AdaptDatabaseState::insertUpdate(), SH_StatementState::onEntry(), SH_BillingCreationStateMachine::SH_BillingCreationStateMachine(), SH_GenericState::SH_GenericState(), SH_InOutState::SH_InOutState(), et SH_ServiceCharging::SH_ServiceCharging()\&.
.SS "SH_InOutState::input () const\fC [virtual]\fP, \fC [inherited]\fP"

.PP
\fBRenvoie:\fP
.RS 4
QVariant 
.RE
.PP

.PP
Définition à la ligne 19 du fichier SH_IOState\&.cpp\&.
.PP
Références SH_InOutState::m_input\&.
.PP
Référencé par SH_InOutState::rawInput(), SH_DateQuestionState::rawInput(), et SH_InOutState::setInput()\&.
.PP
.nf
20 {
21     return m_input;
22 }
.fi
.SS "bool SH_DatabaseContentQuestionState::isAnswerValid (const QVariant &givenAnswer)\fC [virtual]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIgivenAnswer\fP 
.RE
.PP

.PP
Implémente \fBSH_QuestionState\fP\&.
.PP
Définition à la ligne 31 du fichier SH_DatabaseContentQuestionState\&.cpp\&.
.PP
Références m_choices\&.
.PP
.nf
32 {
33     return m_choices\&.isEmpty() || m_choices\&.contains(givenAnswer);
34 }
.fi
.SS "SH_GenericState::isRunning ()\fC [inherited]\fP"

.PP
\fBRenvoie:\fP
.RS 4
bool 
.RE
.PP

.PP
Définition à la ligne 81 du fichier SH_GenericDebugableState\&.cpp\&.
.PP
Références SH_GenericState::m_isRunning\&.
.PP
Référencé par SH_QuestionState::checkValidity(), SH_ConfirmationState::confirmInput(), SH_GenericState::emitGoNext(), SH_InOutState::emitResendInput(), SH_InOutState::emitSendOutput(), SH_InOutState::setInput(), SH_InOutState::setOutput(), et SH_InOutState::setVisibility()\&.
.PP
.nf
82 {
83     return m_isRunning;
84 }
.fi
.SS "SH_GenericState::next ()\fC [signal]\fP, \fC [inherited]\fP"

.PP
Référencé par SH_GenericState::emitGoNext()\&.
.SS "SH_GenericState::onEntry (QEvent *event)\fC [protected]\fP, \fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIevent\fP 
.RE
.PP

.PP
Définition à la ligne 60 du fichier SH_GenericDebugableState\&.cpp\&.
.PP
Références SH_MessageManager::debugMessage(), SH_GenericState::m_isRunning, SH_NamedObject::name(), et SH_GenericState::onTransitionTriggered()\&.
.PP
Référencé par SH_StatementState::onEntry()\&.
.PP
.nf
61 {
62     Q_UNUSED(event);
63     foreach (QAbstractTransition* tr, transitions()) {
64         connect(tr, SIGNAL(triggered()), this, SLOT(onTransitionTriggered()));
65     }
66     m_isRunning = true;
67     this->blockSignals(!m_isRunning);
68     SH_MessageManager::debugMessage(QString("Machine: %1, entered state %2")\&.arg(machine()->objectName())\&.arg(name()));
69 }
.fi
.SS "SH_GenericState::onExit (QEvent *event)\fC [protected]\fP, \fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIevent\fP 
.RE
.PP

.PP
Définition à la ligne 74 du fichier SH_GenericDebugableState\&.cpp\&.
.PP
Références SH_MessageManager::debugMessage(), SH_GenericState::m_isRunning, et SH_NamedObject::name()\&.
.PP
.nf
75 {
76     Q_UNUSED(event);
77     m_isRunning = false;
78     this->blockSignals(!m_isRunning);
79     SH_MessageManager::debugMessage(QString("Machine: %1, exited state %2")\&.arg(machine()->objectName())\&.arg(name()));
80 }
.fi
.SS "SH_InOutState::output () const\fC [virtual]\fP, \fC [inherited]\fP"

.PP
\fBRenvoie:\fP
.RS 4
QString 
.RE
.PP

.PP
Définition à la ligne 47 du fichier SH_IOState\&.cpp\&.
.PP
Références SH_InOutState::m_output\&.
.PP
Référencé par SH_QuestionState::checkValidity(), et SH_InOutState::setOutput()\&.
.PP
.nf
48 {
49     return m_output;
50 }
.fi
.SS "QVariant SH_DatabaseContentQuestionState::rawInput () const\fC [virtual]\fP"

.PP
\fBRenvoie:\fP
.RS 4
QVariant 
.RE
.PP

.PP
Réimplémentée à partir de \fBSH_InOutState\fP\&.
.PP
Définition à la ligne 51 du fichier SH_DatabaseContentQuestionState\&.cpp\&.
.PP
Références SH_QuestionState::givenAnswer(), et m_choices\&.
.PP
Référencé par SH_ServiceCharging::SH_ServiceCharging()\&.
.PP
.nf
52 {
53     return m_choices\&.indexOf(this->givenAnswer());
54 }
.fi
.SS "SH_InOutState::resendInput (QVariantinput)\fC [signal]\fP, \fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIinput\fP 
.RE
.PP

.PP
Référencé par SH_InOutStateMachine::addIOState(), et SH_InOutState::emitResendInput()\&.
.SS "SH_InOutState::sendOutput (QVariantoutput)\fC [signal]\fP, \fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIoutput\fP 
.RE
.PP

.PP
Référencé par SH_InOutStateMachine::addIOState(), et SH_InOutState::emitSendOutput()\&.
.SS "SH_QuestionState::setGivenAnswer (const QVariant &givenAnswer)\fC [virtual]\fP, \fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIgivenAnswer\fP 
.RE
.PP

.PP
Définition à la ligne 56 du fichier SH_QuestionState\&.cpp\&.
.PP
Références SH_QuestionState::checkValidity(), SH_QuestionState::givenAnswer(), et SH_QuestionState::m_givenAnswer\&.
.PP
Référencé par SH_QuestionState::setInput()\&.
.PP
.nf
57 {
58     if(givenAnswer != this->givenAnswer()) {
59         this->m_givenAnswer = givenAnswer;
60         this->checkValidity();
61     }
62 }
.fi
.SS "SH_QuestionState::setInput (const QVariant &input)\fC [virtual]\fP, \fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIinput\fP 
.RE
.PP

.PP
Réimplémentée à partir de \fBSH_InOutState\fP\&.
.PP
Définition à la ligne 38 du fichier SH_QuestionState\&.cpp\&.
.PP
Références SH_QuestionState::givenAnswer(), et SH_QuestionState::setGivenAnswer()\&.
.PP
Référencé par SH_ServiceCharging::SH_ServiceCharging()\&.
.PP
.nf
39 {
40     if(input != this->givenAnswer()) {
41         this->setGivenAnswer(input);
42     }
43 }
.fi
.SS "void SH_DatabaseContentQuestionState::setOutput (const QString &output)\fC [virtual]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIoutput\fP 
.RE
.PP

.PP
Réimplémentée à partir de \fBSH_InOutState\fP\&.
.PP
Définition à la ligne 39 du fichier SH_DatabaseContentQuestionState\&.cpp\&.
.PP
Références displayChoiceList(), m_choices, m_choicesDisplayed, et SH_InOutState::setOutput()\&.
.PP
Référencé par SH_ServiceCharging::SH_ServiceCharging()\&.
.PP
.nf
40 {
41     SH_QuestionState::setOutput(output);
42     if(m_choices\&.size() < 8) {
43         m_choicesDisplayed = true;
44         emit displayChoiceList(m_choices);
45     }
46 }
.fi
.SS "SH_InOutState::setVisibility (boolisVisible)\fC [virtual]\fP, \fC [slot]\fP, \fC [inherited]\fP"

.PP
\fBParamètres:\fP
.RS 4
\fIisVisible\fP 
.RE
.PP

.PP
Définition à la ligne 66 du fichier SH_IOState\&.cpp\&.
.PP
Références SH_GenericState::isRunning(), SH_InOutState::m_isVisible, et SH_InOutState::visibility()\&.
.PP
Référencé par SH_ServiceCharging::SH_ServiceCharging()\&.
.PP
.nf
67 {
68     if(isRunning() && isVisible!=this->visibility()) {
69         m_isVisible = isVisible;
70     }
71 }
.fi
.SS "SH_GenericState::toString ()\fC [virtual]\fP, \fC [inherited]\fP"

.PP
\fBRenvoie:\fP
.RS 4
QString 
.RE
.PP

.PP
Réimplémentée à partir de \fBSH_NamedObject\fP\&.
.PP
Définition à la ligne 23 du fichier SH_GenericDebugableState\&.cpp\&.
.PP
Références SH_NamedObject::toString(), et SH_GenericState::toString()\&.
.PP
Référencé par SH_QuestionState::checkValidity(), SH_DateQuestionState::rawInput(), SH_GenericStateMachine::toString(), et SH_GenericState::toString()\&.
.PP
.nf
24 {
25     QStateMachine* machine = this->machine();
26     SH_GenericState* mach = qobject_cast<SH_GenericState *>(machine);
27     if(mach) {
28         return SH_NamedObject::toString()+ " [in "+mach->toString()+"] ";
29     } else {
30         return SH_NamedObject::toString();
31     }
32 }
.fi
.SS "SH_InOutState::visibility ()\fC [inherited]\fP"

.PP
\fBRenvoie:\fP
.RS 4
bool 
.RE
.PP

.PP
Définition à la ligne 76 du fichier SH_IOState\&.cpp\&.
.PP
Références SH_InOutState::m_isVisible\&.
.PP
Référencé par SH_InOutState::setVisibility()\&.
.PP
.nf
76                                {
77     return m_isVisible;
78 }
.fi
.SH "Documentation des données membres"
.PP 
.SS "QVariantList SH_DatabaseContentQuestionState::m_choices\fC [private]\fP"

.PP
m_choices 
.PP
Définition à la ligne 82 du fichier SH_DatabaseContentQuestionState\&.h\&.
.PP
Référencé par choiceList(), isAnswerValid(), rawInput(), et setOutput()\&.
.SS "bool SH_DatabaseContentQuestionState::m_choicesDisplayed\fC [private]\fP"

.PP
m_choicesDisplayed 
.PP
Définition à la ligne 87 du fichier SH_DatabaseContentQuestionState\&.h\&.
.PP
Référencé par choiceList(), et setOutput()\&.
.SS "QString SH_DatabaseContentQuestionState::m_condition\fC [private]\fP"

.PP
m_condition 
.PP
Définition à la ligne 72 du fichier SH_DatabaseContentQuestionState\&.h\&.
.PP
Référencé par SH_DatabaseContentQuestionState()\&.
.SS "QString SH_DatabaseContentQuestionState::m_field\fC [private]\fP"

.PP
m_field 
.PP
Définition à la ligne 77 du fichier SH_DatabaseContentQuestionState\&.h\&.
.PP
Référencé par SH_DatabaseContentQuestionState()\&.
.SS "QString SH_DatabaseContentQuestionState::m_table\fC [private]\fP"

.PP
m_table 
.PP
Définition à la ligne 67 du fichier SH_DatabaseContentQuestionState\&.h\&.
.PP
Référencé par SH_DatabaseContentQuestionState()\&.

.SH "Auteur"
.PP 
Généré automatiquement par Doxygen pour PreCheck à partir du code source\&.


src/PreCheck/logic/SH_DatabaseContentQuestionState.cpp
#include "SH_DatabaseContentQuestionState.h"
#include "models/SH_SqlDataModel.h"
#include "SH_MessageManager.h"
/*namespace SimplHotel
{*/
/*!
 * \details \~french
 * \fn SH_DatabaseContentQuestionState::DatabaseContentQuestionState
*/
SH_DatabaseContentQuestionState::SH_DatabaseContentQuestionState(QString question, QString name, QString databaseTable, QString tableField, QString databaseCondition, bool noChoiceDisplay, QState *parent) :
    SH_QuestionState(question, name, parent), m_table(databaseTable), m_condition(databaseCondition), m_field(tableField), m_noChoiceDisplay(noChoiceDisplay)
{
    //SH_MessageManager::debugMessage(QString("multiple choice list with datas from %1!").arg(databaseTable));
    SH_SqlDataModel *sqlDatas = new SH_SqlDataModel();
    QStringList fields;
    fields << "ID" << m_field;
    fields.removeDuplicates();
    sqlDatas->fetch(m_table, m_condition, "", fields);
    QVariantMap results = sqlDatas->datas();
    QVariantList idValues = results.values("ID");
    QVariantList fieldsValues = results.values(m_field);
    for(int i = 0; i < idValues.length(); i++) {
        //SH_MessageManager::debugMessage(QString("new choice %1: %2").arg(idValues.at(i).toString()).arg(fieldsValues.at(i).toString()));
        m_choices.insert(idValues.at(i).toInt(), fieldsValues.at(i));
    }
}
/*!
 * \details \~french
 * \fn SH_DatabaseContentQuestionState::isAnswerValid
*/
bool SH_DatabaseContentQuestionState::isAnswerValid(const QVariant &givenAnswer)
{
    return m_choices.isEmpty() || m_choices.contains(givenAnswer);
}
/*!
 * \details \~french
 * \fn SH_DatabaseContentQuestionState::setOutput
 */
void SH_DatabaseContentQuestionState::setOutput(const QString &output)
{
    SH_QuestionState::setOutput(output);
    if(!m_noChoiceDisplay && m_choices.size() < 8) {
        m_choicesDisplayed = true;
        emit displayChoiceList(m_choices);
    }
}
/*!
 * \details \~french
 * \fn SH_DatabaseContentQuestionState::rawInput
*/
QVariant SH_DatabaseContentQuestionState::rawInput() const
{
    return m_choices.indexOf(this->givenAnswer());
}
/*!
 * \details \~french
 * \fn SH_DatabaseContentQuestionState::choiceList
*/
QVariantList SH_DatabaseContentQuestionState::choiceList() const {
    if(m_choicesDisplayed) {
        return m_choices;
    }
    return QVariantList();
}
/*}*/
